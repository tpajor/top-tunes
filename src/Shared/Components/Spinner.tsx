import * as React from 'react';

interface IProps {
  text?: string;
}

class Spinner extends React.PureComponent<IProps> {
  public render() {
    return (
      <div className='container container--list-info'>
        <div className='container container--spinner-icon spinner-rotator'/>
        {this.props.text ? <p className='text text--list-info spinner-dots'>{this.props.text}</p> : ''}
      </div>
    );
  }
}

export default Spinner;
