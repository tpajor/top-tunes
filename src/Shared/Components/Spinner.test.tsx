import * as React from 'react';
import { shallow } from 'enzyme';

import Spinner, { IProps } from './Spinner';

function setup() {
  const props: IProps = {
    text: 'LOADING',
  };

  const wrapper = shallow(<Spinner {...props} />);

  return {
    wrapper,
    props,
  };
}

describe('Spinner component', () => {
  it('should match snapshot if no change was intended', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should not render text if it was not passed in props', () => {
    const { wrapper, props } = setup();

    expect(wrapper.find('p').length).toBe(1);

    wrapper.setProps({ ...props, text: undefined });

    expect(wrapper.find('p').length).toBe(0);
  });
});
