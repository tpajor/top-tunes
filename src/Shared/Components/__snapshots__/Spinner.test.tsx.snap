// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component should match snapshot if no change was intended 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Spinner
    text="LOADING"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="container container--spinner-icon spinner-rotator"
        />,
        <p
          className="text text--list-info spinner-dots"
        >
          LOADING
        </p>,
      ],
      "className": "container container--list-info",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "container container--spinner-icon spinner-rotator",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "LOADING",
          "className": "text text--list-info spinner-dots",
        },
        "ref": null,
        "rendered": "LOADING",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="container container--spinner-icon spinner-rotator"
          />,
          <p
            className="text text--list-info spinner-dots"
          >
            LOADING
          </p>,
        ],
        "className": "container container--list-info",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "container container--spinner-icon spinner-rotator",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "LOADING",
            "className": "text text--list-info spinner-dots",
          },
          "ref": null,
          "rendered": "LOADING",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
